pydantic/_internal/_validators.py

```diff
@@ -308,6 +309,11 @@ def multiple_of_validator(x: Any, multiple_of: Any) -> Any:
         raise TypeError(f"Unable to apply constraint 'multiple_of' to supplied value {x}")
 
 
+# TODO: we have a sort of hacky way of supporting min_length and max_length for path like types as a consequence
+# of prior design that supported out of order metadata injection during core schema generation, which we have since abandoned
+# see: https://github.com/pydantic/pydantic/pull/10846/
```

Comment:
It's notable that we add hacky support for `min_length` and `max_length` below, but we don't support things like `strip_whitespace` or `to_upper` - those are now broken for path like types.